From cfe7742317cbd31899987ec13ed91b21a05cf13e Mon Sep 17 00:00:00 2001
From: Chenhui Zhao <chenhui.zhao@freescale.com>
Date: Fri, 31 Oct 2014 18:25:55 +0800
Subject: [PATCH 061/129] net: gianfar: fix compile error when enabling
 CONFIG_PM

mpc85xx_pmc_set_wake() is only for powerpc platform. For LS1 platform,
no such function. Use a weak function definition to void compile
problem.

Signed-off-by: Chenhui Zhao <chenhui.zhao@freescale.com>
Change-Id: I84ab02a65addabd2f8d7a9681e5f80c7b06c5ad0
Reviewed-on: http://git.am.freescale.net:8181/22821
Tested-by: Review Code-CDREVIEW <CDREVIEW@freescale.com>
Reviewed-by: Yang Li <LeoLi@freescale.com>
Reviewed-by: Zhengxiong Jin <Jason.Jin@freescale.com>
[Xulin: Original patch taken from
Freescale-Linux-SDK-for-LS1021A-IOT-Rev2-v0.4-SOURCE-20150907-yocto.iso]
Signed-off-by: Xulin Sun <xulin.sun@windriver.com>
---
 drivers/net/ethernet/freescale/gianfar_ethtool.c |    8 +++++---
 1 files changed, 5 insertions(+), 3 deletions(-)

diff --git a/drivers/net/ethernet/freescale/gianfar_ethtool.c b/drivers/net/ethernet/freescale/gianfar_ethtool.c
index 50d88d9..c005016 100644
--- a/drivers/net/ethernet/freescale/gianfar_ethtool.c
+++ b/drivers/net/ethernet/freescale/gianfar_ethtool.c
@@ -41,12 +41,14 @@
 #include <linux/phy.h>
 #include <linux/sort.h>
 #include <linux/if_vlan.h>
-#ifdef CONFIG_PM
-#include <sysdev/fsl_soc.h>
-#endif
 
 #include "gianfar.h"
 
+int __attribute__ ((weak)) mpc85xx_pmc_set_wake(struct device *dev, bool enable)
+{
+	return 0;
+}
+
 #define GFAR_MAX_COAL_USECS 0xffff
 #define GFAR_MAX_COAL_FRAMES 0xff
 static void gfar_fill_stats(struct net_device *dev, struct ethtool_stats *dummy,
-- 
1.7.5.4

